src/Query/Filter/TextField.ts

```diff
@@ -3,7 +3,7 @@ import { SubstringMatcher } from '../Matchers/SubstringMatcher';
 import { RegexMatcher } from '../Matchers/RegexMatcher';
 import type { IStringMatcher } from '../Matchers/IStringMatcher';
 import { Field } from './Field';
-import { FilterOrErrorMessage } from './Filter';
+import { type Filter, FilterOrErrorMessage } from './Filter';
```

Comment:
That syntax is actually supported since TypeScript 3.8, and we're requring TypeSript 4.4 in package.json.

Can you check in WebStrom whether you're using the bundled TypeScript or a globally installed TypeScript? Should be somewhere under Settings -> Languages & Frameworks. Maybe either WebStorm or the globally installed TypeScript needs to be updated.

Reply:
Thank you for the info.

These are my typescript settings in WebStorm:
![image](https://user-images.githubusercontent.com/4840096/192004493-407537fc-8092-4a1c-ab01-084b0ca703ef.png)

Reply:
Here is what the WebStorm docs say about TypeScript checks:
https://www.jetbrains.com/help/webstorm/typescript-support.html#ws_ts_restart_ts_service

 it something to do with
https://github.com/obsidian-tasks-group/obsidian-tasks/blob/ca25711dc1f237663a29e99eee368a949b3b7cf1/tsconfig.json

Reply:
Your TypeScript language service is up to date and the config looks ok, too. Seems it is caused not by the TypeScript language service, but by the auto import feature of WebStorm - see [here](https://youtrack.jetbrains.com/issue/WEB-43269). This is something we cannot change, so I will revert to the old syntax to make this more compatible.

Reply:
Thank you, I doubt I would ever have figured that out.